### PR TITLE
AC-873 Refactored the repository section

### PR DESCRIPTION
**Pull request by LuGO0**
## Description of what I changed
Refactored the Unit tests as well as the code to encorporate an abstract hierarchy.
## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-873

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [x] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by LuGO0** [comment]
@rishabh-997 please review !!

**Comment by LuGO0** [comment]
@rishabh-997 on a side-note is codeCoverage tool working?

File: openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/ProviderDashboardPresenterTest.java [hunk 1]
```diff
@@ -39,12 +39,13 @@
     private OpenMRSLogger openMRSLogger;
     @Mock
     private OpenMRS openMRS;
+
```
Comment (HerbertYiga) [inline openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/ProviderDashboardPresenterTest.java:42]: you can remove this added line too

Comment (LuGO0) [reply openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/ProviderDashboardPresenterTest.java:42]: Sure @HerbertYiga I will remove this line once my PR gets reviewed 👍

File: openmrs-client/src/main/java/org/openmrs/mobile/api/repository/BaseRepository.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    public BaseRepository(){
+        this.openMrs = OpenMRS.getInstance();
+        this.restApi = RestServiceBuilder.createService(RestApi.class);
+        db = AppDatabase.getDatabase(openMrs);
+        workManager = WorkManager.getInstance(openMrs);
+        logger = new OpenMRSLogger();
+    }
+
+    // used in Unit tests
```
Comment (rishabh-997) [inline openmrs-client/src/main/java/org/openmrs/mobile/api/repository/BaseRepository.java:41]: Remove this comment,

File: openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/ProviderDashboardPresenterTest.java [hunk 3]
```diff
@@ -39,12 +39,13 @@
...
     private ProviderDashboardPresenter providerDashboardPresenter;
     private ProviderRepository providerRepository;
 
     @Before
     public void setup() {
-        this.providerRepository = new ProviderRepository(restApi);
+        this.providerRepository = new ProviderRepository(openMRS,restApi);
```
Comment (rishabh-997) [inline openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/ProviderDashboardPresenterTest.java:48]: add a space after the comma.

File: openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/ProviderManagerDashboardPresenterTest.java [hunk 4]
```diff
@@ -75,7 +75,7 @@ public void setUp() {
         providerList = Arrays.asList(providerOne, providerTwo);
         providerLiveData.postValue(providerList);
 
-        this.providerRepository = new ProviderRepository(restApi);
+        this.providerRepository = new ProviderRepository(openMRS,restApi);
```
Comment (rishabh-997) [inline openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/ProviderManagerDashboardPresenterTest.java:78]: add space after the comma.

File: openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/FormAdmissionPresenterTest.java [hunk 5]
```diff
@@ -104,7 +104,7 @@ public void setUp() {
         resourceList = Arrays.asList(resourceOne, resourceTwo);
         providerLiveData.postValue(providerList);
 
-        this.providerRepository = new ProviderRepository(restApi);
+        this.providerRepository = new ProviderRepository(openMRS,restApi);
```
Comment (rishabh-997) [inline openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/FormAdmissionPresenterTest.java:107]: add a space.

**Review by rishabh-997 — CHANGES_REQUESTED** [review]
Yo.

File: openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/FormAdmissionPresenterTest.java (hunk 5 above)
Comment (LuGO0) [reply openmrs-client/src/test/java/org/openmrs/mobile/test/presenters/FormAdmissionPresenterTest.java:107]: Oops I forgot reformatting doing it 👍

**Review by rishabh-997 — APPROVED** [review]

**Pull request merged**